website/src/releases/2020-10/release-v17.mdx

```diff
@@ -0,0 +1,233 @@
+
+```diff
+const { loading, data, error } = useQuery(MyQuery, {
+- variables: {
++ context: {
+    target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
+  },
+});
+```
+
+The **request context** options include:
+
+- `target`
+- `projectKey`
+
+<Info>
+
+Passing the request metadata to the `variables` object still works for backwards compatibility but it is recommended to use the `context` object instead.
+
+</Info>
+
+## Enforcing a valid context object
+
+The recommended and preferred way of passing request context is to use the `context` object.
+To improve the TypeScript support and auto-completion for the **request context** options, the `@commercetools-frontend/application-shell` package now exports new React hooks. Note that autocompletion is possible, even if you are not using TypeScript.
+
+- `useMcQuery`
+- `useMcLazyQuery`
+- `useMcMutation`
+
+These hooks are thin wrappers around the original Apollo hooks and have the same API, with a minor difference. Namely, the `context` object is properly typed to conform with the Merchant Center **request context** instead of `any`.
+
+```diff
+-import { useQuery } from '@apollo/client/react';
++import { useMcQuery } from '@commercetools-frontend/application-shell';
+
+-const { loading, data, error } = useQuery(MyQuery, {
++const { loading, data, error } = useMcQuery(MyQuery, {
+  context: {
+    target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
+  },
+});
+```
+
+# Changes about the `mc-scripts` CLI
+
+## Removing deprecated options from `compile-html`
+
+Previously deprecated CLI options such as `--env`, `--csp`, `--headers`, have been removed.
+
+Additionally, the CLI flag `--use-local-assets` has been removed as well. The default behavior of `mc-scripts compile-html` now is to compile the assets locally, which is the only reasonable thing to do.
+
+When running the `mc-scripts compile-html` command, the `index.html` is compiled for production usage and it lives in the `public` folder, together with the other static assets. This is all you need to deploy your application.
+You can decide to [deploy the Custom Application statically to one of the popular cloud providers](/deployment/compiling-a-custom-application#deployment), or serve the files on your own using a static server.
+
+For example, to run locally the Custom Application using the production bundles:
+
+```console
+NODE_ENV=production MC_APP_ENV=development dotenv -- \
+  mc-scripts compile-html \
+  --transformer @commercetools-frontend/mc-dev-authentication/transformer-local.js
+
+mc-scripts serve
+```
+
+## Removing the `mc-http-server` package
+
+The `@commercetools-frontend/mc-http-server` package has been deprecated and won't be published anymore.
+
+With the usage of the `compile-html` command there is no need to have a pre-configured HTTP server anymore. If you are using this package, we recommend to use any other HTTP server package to serve your static files.
```

Comment:
Maybe we can recommend one.

Reply:
You mean like express?

Reply:
😄 I don't mind. Express or serve or any package. We don't have to was just a thought.